src/ofxstatement/statement.py

```diff
@@ -32,6 +32,18 @@
     "CREDITLINE",   # Line of credit
 ]
 
+# 2020-03-12 Gert-Jan Paulissen
+#
+# Note: generation of an id
+#
+# A list of statement line id's to ensure uniqueness in an OFX.
+# Recreated each time a Statement is initialised.
+# Not the most elegant solution but necessary since the generation of an id
+# occurs in the context of a statement line and then the statement is not
+# accessible.
+# The need to clear _ids is due to testing several creations of a Statement.
+_ids = None
```

Comment:
Yeah, global is not great. I think we can improve this a bit.

A bit of background: the problem with these transaction ids is that they are not only have to be unique, they also have to stay the same for the same transaction every time you generate the statement. I.e. generating random ids will not work, even though they will be unique, GnuCash or beancount will recognize these transaction as "new" if you happen to generate and import the same statement twice or import two statements with overlapping periods.

The current function (`generate_transaction_id()`) is deterministic, but does not necesserily generate an unique id, as you correctly pointed out. I propose providing another function, like this:

```python
def generate_unique_transaction_id(stmt_line, generated)`
```

Where `generated` would be a set of previously generated transactions. The function would modify the set and return the generated transaction too.

The algorithm we can use to ensure uniqueness: 
1. generate usual id with `generate_transaction_id`
2. if generated id is in `generated`, add 1 to it. Repeat until unique transaction found
3. Add it to `generated` set
4. Return it

Then plugins can use it the way they like.

Reply:
Funny, this was actually the first thing I tried: add something to the unique id. However, the financial tool (beancount-import) I use got a bit confused when there are two identical statement lines with just a different id. So I just added ' #' + <current record number> to the memo field and recalculated the id. That worked. 

In short, this is a good idea and generated should be a set (good suggestion). But I would suggest also another parameter to handle my use case, called strict or something like that. So either it raises an exception (if strict) or it keeps adding a 1 till it is unique (not strict).

Reply:
I did not understand it too, but with the current implementation I got my problem solved. The counter is returned and I can decide to adapt the memo field. And it will be deterministic.